filters/include/pcl/filters/impl/crop_hull.hpp

```diff
@@ -113,72 +92,83 @@ pcl::CropHull<PointT>::getHullCloudRange ()
       if (pt[i] > cloud_max[i]) cloud_max[i] = pt[i];
     }
   }
-  
+
   return (cloud_max - cloud_min);
 }
 
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 template<typename PointT> template<unsigned PlaneDim1, unsigned PlaneDim2> void 
-pcl::CropHull<PointT>::applyFilter2D (PointCloud &output)
+pcl::CropHull<PointT>::applyFilter2D (std::vector<int> &indices)
 {
-  for (std::size_t index = 0; index < indices_->size (); index++)
+  Eigen::Vector4f const minPt = crop_box_.getMin();
+  Eigen::Vector4f const maxPt = crop_box_.getMax();
+  std::vector<bool> cropInlierMask(input_->size(), true);
+  for (const index_t idx : *indices_)
   {
-    // iterate over polygons faster than points because we expect this data
-    // to be, in general, more cache-local - the point cloud might be huge
-    std::size_t poly;
-    for (poly = 0; poly < hull_polygons_.size (); poly++)
+    Eigen::Vector3f pt = input_->points[idx].getVector3fMap();
+    if (pt[PlaneDim1] < minPt[PlaneDim1] || pt[PlaneDim1] > maxPt[PlaneDim2] ||
```

Comment:
Is it really `pt[PlaneDim1] > maxPt[PlaneDim2]`, shouldn't it be `pt[PlaneDim1] > maxPt[PlaneDim]`?

Reply:
fix

also modified test functions to detect this bug